pkg/upgrade/machine_creator.go

```diff
@@ -88,32 +103,25 @@ func (n *MachineCreator) NewMachine(source *clusterapiv1alpha1.Machine) (*cluste
 	// have to clear this out so the new machine can get its own provider id set
 	newMachine.Spec.ProviderID = nil
 
-	// assume the original name is controlplane-<index> or controlplane-<index>-<timestamp>
-	// let's set the new name to controlplane-<index>-<timestamp>
-	nameParts := strings.Split(source.Name, "-")
-	if len(nameParts) < 2 {
-		return nil, nil, errors.Errorf("machine name %q does not match expected format <name>-<index> or <name>-<index>-<timestamp>", source.Name)
-	}
-	newMachine.Name = fmt.Sprintf("%s-%s-%d", nameParts[0], nameParts[1], time.Now().Unix())
+	newMachine.Name = name
 
 	if n.MachineOptions.ImageField != "" && n.MachineOptions.ImageID != "" {
 		if err := updateMachineSpecImage(&newMachine.Spec, n.MachineOptions.ImageField, n.MachineOptions.ImageID); err != nil {
 			return nil, nil, err
 		}
 	}
 
-	newMachine.Spec.Versions.ControlPlane = n.MachineOptions.DesiredVersion.String()
-	newMachine.Spec.Versions.Kubelet = n.MachineOptions.DesiredVersion.String()
+	desiredVersion := n.MachineOptions.DesiredVersion.String()
+	newMachine.Spec.Version = &desiredVersion
 
 	n.log.Info("Creating new machine", "name", newMachine.Name)
 
-	createdMachine, err := n.machineCreator.Create(newMachine)
-	if err != nil {
+	if err := n.managementClient.Create(context.TODO(), newMachine); err != nil {
```

Comment:
If you make this change, are we still using `n.machineCreator` anywhere?

Reply:
nope. good catch